### PR TITLE
docs: maintainership prep

### PR DESCRIPTION
**Pull request by nsprenkle**
Add missing docs & config to align with maintainership guidelines.

JIRA: https://2u-internal.atlassian.net/browse/AU-1020

### CODE REVIEW

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - url: "ora-grading.stage.edx.org"
+      title: "Stage Site"
+      icon: "Web"
+spec:
+  owner: group:content-aurora
+  type: 'website'
+  lifecycle: 'production'
+  dependsOn:
+  - 'edx-ora2'
+  - 'edx-platform'
```
Comment (schenedx) [inline catalog-info.yaml]: #NIT should we have a newline here?

**Review by schenedx — APPROVED** [review]
👍 LGTM

File: catalog-info.yaml (hunk 1 above)
Comment (nsprenkle) [reply catalog-info.yaml]: Yes

**Pull request merged**